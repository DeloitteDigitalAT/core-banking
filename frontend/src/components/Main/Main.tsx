import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { Header, Tabs, Form } from "../UI";

const Main: React.FC = () => {
  const [tabsElements, setTabsElements] = useState([
    { name: "Search by Name", id: 1, active: true },
    { name: "Search by Application ID", id: 2, active: false },
  ]);
  const history = useHistory();
  const [formName, setFormName] = useState([
    {
      id: 1,
      type: "input",
      label: "First Name",
      placeholder: "John",
      value: "",
      required: false,
      readonly: false,
      error: false,
      divClass: "col-span-6 max-w-sm",
    },
    {
      id: 2,
      type: "input",
      label: "Last Name",
      placeholder: "Doe",
      value: "",
      required: true,
      readonly: false,
      error: false,
      divClass: "col-span-6 max-w-sm",
    },
    {
      id: 3,
      type: "button",
      text: "Search",
      handler: () => {},
      divClass: "col-span-6 max-w-sm",
    },
  ]);
  const [formID, setFormID] = useState([
    {
      id: 1,
      type: "input",
      label: "Application ID",
      placeholder: "123456",
      value: "",
      required: true,
      readonly: false,
      error: false,
      divClass: "col-span-6 max-w-sm",
    },
    {
      id: 3,
      type: "button",
      text: "Search",
      handler: () => {
        history.push(`/application/${formID[0].value}`);
      },
      divClass: "col-span-6",
    },
  ]);
  const activeTabId = tabsElements.find(el => el.active)?.id;
  return (
    <div className="flex-grow">
      <Header>Search customer application</Header>
      <Tabs elements={tabsElements} setElements={setTabsElements} />
      {activeTabId === 1 && <Form elements={formName} setForm={setFormName} />}
      {activeTabId === 2 && <Form elements={formID} setForm={setFormID} />}
    </div>
  );
};

export default Main;
