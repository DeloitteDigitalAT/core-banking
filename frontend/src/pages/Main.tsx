import React from "react";
import Main from "../components/Main/Main";

const PageMain: React.FC = () => {
  return <Main />;
};

export default PageMain;
