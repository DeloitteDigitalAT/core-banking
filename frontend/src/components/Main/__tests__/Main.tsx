import React from "react";
import renderer from "react-test-renderer";
import Main from "../Main";

describe("Main component", () => {
  test("snapshot renders", () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
