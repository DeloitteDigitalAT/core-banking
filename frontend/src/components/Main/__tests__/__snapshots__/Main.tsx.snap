// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component snapshot renders 1`] = `
<div
  className="flex-grow"
>
  <div
    className="text-xl py-4 mb-0 text-gray-800"
  >
    Search customer application
  </div>
  <ul
    className="list-reset flex border-b mb-4"
  >
    <li
      className="-mb-px mr-1"
    >
      <div
        className="bg-white inline-block py-2 px-4 text-sm  cursor-pointer font-semibold text-gray-800 border-l border-t border-r rounded-t"
        onClick={[Function]}
      >
        Search by Name
      </div>
    </li>
    <li
      className="-mb-px mr-1"
    >
      <div
        className="bg-white inline-block py-2 px-4 text-sm  cursor-pointer text-gray-400 hover:text-gray-800"
        onClick={[Function]}
      >
        Search by Application ID
      </div>
    </li>
  </ul>
  <div
    className="grid grid-cols-6"
  >
    <div
      className="col-span-6 max-w-sm"
    >
      <label
        className="block mt-4"
      >
        <span
          className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
        >
          First Name
        </span>
        <input
          className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  "
          onChange={[Function]}
          placeholder="John"
          readOnly={false}
          value=""
        />
      </label>
    </div>
    <div
      className="col-span-6 max-w-sm"
    >
      <label
        className="block mt-4"
      >
        <span
          className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
        >
          Last Name
           *
        </span>
        <input
          className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  "
          onChange={[Function]}
          placeholder="Doe"
          readOnly={false}
          value=""
        />
      </label>
    </div>
    <div
      className="col-span-6 max-w-sm"
    >
      <button
        className="bg-gradient-to-b from-green-400 via-green-500 to-green-600 border-green-500 hover:from-green-600 hover:via-green-700 hover:to-green-800 text-sm border text-white rounded-md px-6 py-2 my-4 transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
